src/react-native-animated-charts/src/charts/linear/ChartPath.tsx

```diff
@@ -0,0 +1,361 @@
+    }, []);
+
+    useEffect(() => {
+      if (currentPath?.path === previousPath?.path) {
+        return;
+      }
+
+      runOnUI(() => {
+        'worklet';
+
+        // setOriginData(paths[1]);
+
+        if (progress.value !== 0 && progress.value !== 1) {
+          cancelAnimation(progress);
+        }
+
+        progress.value = 0;
+
+        // this stores an instance of d3-interpolate-path on worklet side
+        // it means that we don't cross threads with that function
+        // which makes it super fast
+        if (previousPath && currentPath) {
+          const d3Interpolate = requireOnWorklet('d3-interpolate-path');
+
+          interpolatorWorklet().value = d3Interpolate.interpolatePath(
+            previousPath.path,
+            currentPath.path
+          );
+
+          progress.value = withDelay(
+            Platform.OS === 'ios' ? 0 : 100,
+            withTiming(1, timingAnimationConfig || timingAnimationDefaultConfig)
+          );
+        } else {
+          interpolatorWorklet().value = undefined;
+          progress.value = 1;
+        }
+      })();
+    }, [currentPath, previousPath]);
+
+    useAnimatedReaction(
+      () => ({ x: translationX.value, y: translationY.value }),
+      values => {
+        if (
+          !currentPath ||
+          !currentPath.parsed ||
+          progress.value === 0 ||
+          values.x === null ||
+          values.y === null
+        ) {
+          return;
+        }
+
+        const yForX = redash.getYForX(currentPath.parsed, Math.floor(values.x));
+
+        if (yForX !== null) {
+          positionY.value = yForX;
+        }
+
+        positionX.value = values.x;
+
+        const index = least(currentPath.points.length, i => {
+          if (typeof i === 'undefined') {
+            return 0;
+          }
+
+          return Math.hypot(currentPath.points[i].x - Math.floor(values.x));
+        });
+
+        setOriginData(currentPath, index);
+      },
+      [currentPath]
+    );
+
+    const animatedProps = useAnimatedProps(() => {
+      const props: PathProps & ViewProps = {};
+
+      if (!currentPath) {
+        return {
+          d: '',
+        };
+      }
+
+      props.d = interpolatorWorklet().value
+        ? interpolatorWorklet().value(progress.value)
+        : currentPath.path;
+
+      props.strokeWidth =
+        pathOpacity.value *
+          (Number(strokeWidth) - Number(selectedStrokeWidth)) +
+        Number(selectedStrokeWidth);
+
+      if (Platform.OS === 'ios') {
+        props.style = {
+          opacity: pathOpacity.value * (1 - selectedOpacity) + selectedOpacity,
+        };
+      }
+
+      return props;
+    }, [currentPath]);
+
+    const onGestureEvent = useAnimatedGestureHandler<LongPressGestureHandlerGestureEvent>(
+      {
+        onStart: event => {
+          // WARNING: the following code does not run on using iOS, but it does on Android.
+          // I use the same code from onActive
+          state.value = event.state;
```

Comment:
move this to some worklet, idk, i don't line copy-pasting

Reply:
I use the state shared value for compatibility reasons, I don't use it at all in the library. If I move this into a worklet - then it'll be copypasting of same worklet call which. Calling worklet is still a bit slower than not calling it.